src/client-side-encryption/state_machine.ts

```diff
@@ -255,9 +282,12 @@ export class StateMachine {
         }
```

Comment:
I can't comment on it because it isn't in the diff.  Can we add a TODO to line 179 linking this ticket? [DRIVERS-2671](https://jira.mongodb.org/browse/DRIVERS-2671)

We need to make automatic kms credential refresh respect CSOT too, but that drivers work has not been completed.

Reply:
Done!